sample-app/src/components/LocationBias.tsx

```diff
@@ -41,20 +33,17 @@ export default function LocationBias(props: Props) {
         : locationBias?.method === LocationBiasMethod.Device ? '(based on your device)'
           : '';
 
-  function handleGeolocationClick() {
-    if ('geolocation' in navigator) {
-      navigator.geolocation.getCurrentPosition((position) => {
-        answersActions.setUserLocation({
-          latitude: position.coords.latitude,
-          longitude: position.coords.longitude,
-        })
-        props.isVertical
-          ? answersActions.executeVerticalQuery()
-          : answersActions.executeUniversalQuery()
-      },
-      (err) => console.error(err),
-      geolocationOptions);
+  async function handleGeolocationClick() {
+    try {
+      const position = await getUserLocation(geolocationOptions);
+      answersActions.setUserLocation({
+        latitude: position.coords.latitude,
+        longitude: position.coords.longitude,
+      });
+    } catch (e) {
```

Comment:
Would it be helpful to log an error about failing to get the user's location?

Reply:
yeah, that error is log in getUserLocation on geolocation API failure, in the case that geolocation API is not accessible, it would reject with a message, where locationbias would conssor error out in line 44